src/os/bluestore/BlueStore.h

```diff
@@ -551,6 +551,46 @@ class BlueStore : public ObjectStore,
     vector<OnodeRef> wal_op_onodes;
 
     interval_set<uint64_t> allocated, released;
+    struct volatile_statfs{
+      enum {
+        STATFS_ALLOCATED = 0,
+        STATFS_STORED,
+        STATFS_COMPRESSED_ORIGINAL,
+        STATFS_COMPRESSED,
+        STATFS_COMPRESSED_ALLOCATED,
+        STATFS_LAST
+      };
+      int64_t values[STATFS_LAST];
+      volatile_statfs() {
+        memset( this, 0, sizeof(volatile_statfs));
+      }
+      void reset() {
+        *this = volatile_statfs();
+      }
+      int64_t& allocated() {
+        return values[STATFS_ALLOCATED];
+      }
+      int64_t& stored() {
+        return values[STATFS_STORED];
+      }
+      int64_t& compressed_original() {
+        return values[STATFS_COMPRESSED_ORIGINAL];
+      }
+      int64_t& compressed() {
+        return values[STATFS_COMPRESSED];
+      }
+      int64_t& compressed_allocated() {
+        return values[STATFS_COMPRESSED_ALLOCATED];
+      }
+      bool is_empty() {
+        return allocated() == 0 &&
+          stored() == 0 &&
+          compressed() == 0 &&
+          compressed_original() == 0 &&
+          compressed_allocated();
```

Comment:
compressed_allocated() == 0

This might be simpler as a loop over the int array, tho, to avoid needing an update if this struct changes